docs/assessment.md

```diff
@@ -350,6 +351,518 @@ CREATE TABLE [IF NOT EXISTS] table_name
+
+### AF302.3 - Arbitrary Java
+
+The `._jdf` is used to execute arbitrary Java code.
+
+[[back to top](#migration-assessment-report)]
+
+### AF302.4 - Arbitrary Java
+
+The `._jcol` is used to execute arbitrary Java code.
+
+[[back to top](#migration-assessment-report)]
+
+### AF302.5 - Arbitrary Java
+
+The `._jvm` is used to execute arbitrary Java code.
+
+[[back to top](#migration-assessment-report)]
+
+### AF302.6 - Arbitrary Java
+
+The `._jvm.org.apache.log4j` is used to execute arbitrary Java code.
+
+[[back to top](#migration-assessment-report)]
+
+### AF303.1 - Java UDF
+
+The `spark.udf.registerJavaFunction` is used to register a Java UDF.
+
+[[back to top](#migration-assessment-report)]
+
+### AF304.1 - JDBC datasource
+
+The `spark.read.format("jdbc")` pattern was found and is used to read data from a JDBC datasource.
+
+Accessing third-party databases—other than MySQL, PostgreSQL, Amazon Redshift, Snowflake, Microsoft SQL Server, Azure Synapse (SQL Data Warehouse) and Google BigQuery will require additional permissions on a shared cluster if the user is not a workspace admin. This is due to the drivers not guaranteeing user isolation, e.g., as the driver writes data from multiple users to a widely accessible temp directory.
+
+Workaround:
+Granting ANY FILE permissions will allow users to access untrusted databases. Note that ANY FILE will still enforce ACLs on any tables or external (storage) locations governed by Unity Catalog.
+This requires DBR 12.2 or later (DBR 12.1 or before is blocked on the network layer)
+
+[[back to top](#migration-assessment-report)]
+
+### AF305.1 - boto3
+
+The `boto3` library is used.
+
+Instance profiles (AWS) and service principals/managed identity (Azure) are not supported from the Python/Scala REPL or UDFs, e.g. using boto3 or s3fs, Instance profiles are only set from init scripts and (internally) from Spark. 
+
+**Workarounds**
+For accessing cloud storage (S3, ADLS, GCS), use storage credentials and external locations. 
+For accessing non-storage cloud services (e.g., AWS secrets manager, etc), use
+
+
+(AWS) Consider other ways to authenticate with boto3, e.g., by passing credentials from Databricks secrets directly to boto3 as a parameter, or loading them as environment variables. This page contains more information. Please note that unlike instance profiles, those methods do not provide short-lived credentials out of the box, and customers are responsible for rotating secrets according to their security needs.
+
+
+(Azure) Consider creating a Databricks secrets scope backed by Azure vault, and authenticate to external systems using secrets from that secrets scope
+
+
+(GCP) Some customers have used an external secrets store (like Hashicorp vault) and loaded credentials upon cluster startup into the cluster’s environment variables. Please note that those are usually not short-lived credentials, and customers are responsible for securing and rotation secrets according to their security needs.
```

Comment:
boto3 is only used for AWS, so this advice for Azure & GCP wouldn't really apply

Reply:
Good point. Thanks!